x-pack/plugin/security/qa/multi-cluster/src/javaRestTest/java/org/elasticsearch/xpack/remotecluster/RemoteClusterSecuritySpecialUserIT.java

```diff
@@ -74,12 +79,32 @@ public class RemoteClusterSecuritySpecialUserIT extends AbstractRemoteClusterSec
                 }
                 return (String) API_KEY_MAP_REF.get().get("encoded");
             })
+            .apply(() -> CLIENT_AUTH_CONFIG_PROVIDERS.get().v2())
             .build();
     }
 
+    // Randomly enable client authentication for remote cluster connection
+    public static void randomClientAuthenticationConfig() {
+        if (randomBoolean()) {
+            CLIENT_AUTH_CONFIG_PROVIDERS.set(
```

Comment:
This is for my own understanding: what exactly necessitates `set` here (over `setPlain`)? `outerRule` should be invoked exactly once, without racing, or no?

Reply:
It's not about racing since both methods do not care what the existing value is. The `set` method has memory effect that guarantees the change to be visible in all threads while `setPlain` does not. Often times, it is simply safer to use the `set` method unless we are 100% sure about thread usages and really care about the small overhead comes with the visibilty guarantee and I don't think it applies to tests here.